src/encoded/static/components/item-pages/MicroscopeConfigurationView.js

```diff
@@ -0,0 +1,542 @@
+                    <i className={"icon icon-fw fas icon-" + (!isFullscreen ? 'expand' : 'compress')}/>
+                </button>
+            );
+        }
+        return null;
+    }
+
+    onSaveMicroscope(microscope, complete) {
+        const { href, context } = this.props;
+        const { modal } = this.state;
+
+        if (!this.havePermissionToEdit()) {
+            // I guess would also get caught in ajax error callback.
+            throw new Error('No edit permissions.');
+        }
+
+        if (modal == null && context.status && typeof context.status === 'string' &&
+            (context.status === 'released' || context.status === 'released to project')) {
+            this.setState({
+                'modal': (
+                    <ConfirmModal handleConfirm={this.handleSave} handleCancel={this.handleModalCancel}
+                        confirmButtonText="Save" cancelButtonText="Cancel" modalTitle="Confirm Save">
+                        You are overwriting a Microscope Configuration Item that was previously shared with public. Are you sure?
+                        <br />Note that you can also clone this display and share the new copy.
+                    </ConfirmModal>
+                )
+            });
+            return;
+        }
+
+        // We're updating this object's microscope.
+        const payload = {
+            'title': microscope.Name,
+            'microscope': microscope
+        };
+
+        this.setState({ 'saveLoading': true, 'modal': null }, () => {
+            ajax.load(
+                href,
+                (resp) => {
+                    // Success callback... maybe update state.originalViewConfigString or something...
+                    // At this point we're saved maybe just notify user somehow if UI update re: state.saveLoading not enough.
+                    Alerts.queue({
+                        'title': "Saved " + microscope.Name,
+                        'message': "This Microscope Configuration Item has been updated with the current viewport. This may take a few minutes to take effect.",
+                        'style': 'success'
+                    });
+                    this.setState({ 'saveLoading': false });
+                },
+                'PATCH',
+                () => {
+                    // Error callback
+                    Alerts.queue({
+                        'title': "Failed to save display.",
+                        'message': "Sorry, can you try to save again?",
+                        'style': 'danger'
+                    });
+                    this.setState({ 'saveLoading': false });
+                },
+                JSON.stringify(payload)
+            );
+        });
+
+        if (complete) {
+            complete(microscope.Name);
+        }
+    }
+
+    render(){
+        const { schemas, context, windowWidth, windowHeight } = this.props;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['schemas' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4271272)